slotfills/sections/ajax-autocomplete.tsx

```diff
@@ -0,0 +1,247 @@
+import React, { useState, useEffect } from 'react';
+import { Spinner } from '@wordpress/components';
+import Downshift from 'downshift';
+import { __ } from '@wordpress/i18n';
+import apiFetch from '@wordpress/api-fetch';
+import { isArray } from 'lodash';
+
+interface AjaxAutocompleteProps {
+  ajaxAction: string;
+  label?: string;
+  initialValue?: string;
+  setValue: Function;
+  showEditLink?: boolean;
+}
+
+interface Post {
+  label: string;
+  value: string;
+}
+
+interface FormDataProps {
+  [key: string]: string;
+}
+
+export default function AjaxAutocomplete({
+  ajaxAction,
+  initialValue, // eslint-disable-line @typescript-eslint/no-unused-vars
+  label,
+  setValue,
+  showEditLink = false,
+}: AjaxAutocompleteProps) {
+  const [foundPosts, setFoundPosts] = useState<Post[]>([]);
+  const [searchText, setSearchText] = useState('');
+  const [working, setWorking] = useState(false);
+  const threshold = 3;
+  const {
+    ajaxurl,
+    fm_search: {
+      nonce: fmSearchNonce = '',
+    } = {},
+  } = (window as any);
+
+  const onChange = (newValue:Post) => {
+    const selectedValue = newValue ? newValue.value : null;
+    setValue(selectedValue);
+  };
+
+  /**
+   * Handles a post selection. Calls `onChange` from props with
+   * the found post object.
+   * @param {object} foundPost - The selected post.
+   */
+  const handlePostSelect = (foundPost: Post) => {
+    // Call the passed onChange function from props with the post object.
+    onChange(foundPost);
+
+    // Reset the internal state.
+    setFoundPosts([]);
+    setSearchText('');
+  };
+
+  useEffect(() => {
+    /**
+     * Loads found posts for the given post type and search text from the API.
+     * @param {string} searchText - The text string to use when searching.
+     */
+    const loadById = (id: number) => {
+      setWorking(true);
+
+      apiFetch({
+        path: `/wp/v2/search?include=${id}`,
+      })
+        .then((result) => {
+          if (!isArray(result)) {
```

Comment:
The native `Array.isArray` could also be an option here if you didn't want to use lodash.

Reply:
@efuller thanks - that was my intention, but I forgot `Array.` and VSCode auto imported it for me.